cmd/server-main.go

```diff
@@ -639,37 +636,34 @@ func serverMain(ctx *cli.Context) {
 		logFatalErrs(err, Endpoint{}, true)
 	}
 
+	if !globalDisableFreezeOnBoot {
+		// Freeze the services until the bucket notification subsystem gets initialized.
+		bootstrapTrace("freezeServices")
+		freezeServices()
```

Comment:
can we freeze the services before setting the object layer?  because, the s3 apis will start serving immediately as soon as the object layer is set.

Reply:
we can @Praveenrajmani